server/py/framework/db/sqldb/db.py

```diff
@@ -2076,10 +2076,11 @@ def update_function(
             struct = function.struct
             for key, val in updates.items():
                 update_in(struct, key, val)
-            function.kind = struct.pop("kind", None)
+            function.kind = (
+                struct.pop("kind", None) if not function.kind else function.kind
+            )
             function.struct = struct
             self._upsert(session, [function])
-            return function.struct
```

Comment:
why not return it?

Reply:
@TomerShor just use `get_function` then, it aligns flow with store_function which also doesn't return function.